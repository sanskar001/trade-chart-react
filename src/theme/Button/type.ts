type ButtonType = "button" | "submit" | "reset";

type ButtonClassType = "btn-outlined" | "";

type ClickHandlerType = React.MouseEventHandler<HTMLButtonElement>;

type PropsType =
  | string
  | boolean
  | undefined
  | ButtonType
  | React.ReactNode
  | ClickHandlerType
  | ButtonClassType
  | React.CSSProperties;

export interface ButtonProps {
  children: React.ReactNode;
  type?: ButtonType;
  onClick?: ClickHandlerType;
  btnClass?: ButtonClassType;
  className?: string;
  style?: React.CSSProperties;
  disabled?: boolean;
  [key: string]: PropsType;
}
