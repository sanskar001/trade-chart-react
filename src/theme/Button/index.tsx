import React from "react";
import { formatClass } from "../../util/formatClass";

type ButtonType = "button" | "submit" | "reset";

type ButtonClassType = "btn-outlined" | "";

type ClickHandlerType = React.MouseEventHandler<HTMLButtonElement>;

type PropsType =
  | string
  | boolean
  | undefined
  | ButtonType
  | React.ReactNode
  | ClickHandlerType
  | ButtonClassType
  | React.CSSProperties;

interface ButtonProps {
  children: React.ReactNode;
  type?: ButtonType;
  onClick?: ClickHandlerType;
  btnClass?: ButtonClassType;
  className?: string;
  style?: React.CSSProperties;
  disabled?: boolean;
  [key: string]: PropsType;
}

const Button: React.FC<ButtonProps> = ({
  children,
  type = "button",
  onClick,
  btnClass = "",
  className = "",
  style,
  disabled = false,
  ...props
}) => {
  const buttonClass = formatClass(`btn ${btnClass} ${className}`);

  return (
    <button
      type={type}
      style={style}
      className={buttonClass}
      disabled={disabled}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
