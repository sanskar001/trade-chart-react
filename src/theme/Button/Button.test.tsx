import renderer from "react-test-renderer";
import Button from ".";

const clickHandler = jest.fn();

it("renders correctly when there is default button", () => {
  const tree = renderer.create(<Button>Default</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is button with onclick handle", () => {
  const tree = renderer
    .create(<Button onClick={clickHandler}>Default</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is submit button", () => {
  const tree = renderer.create(<Button type="submit">Submit</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is outlined button", () => {
  const tree = renderer
    .create(<Button btnClass="btn-outlined">Outlined</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is disabled outlined button", () => {
  const tree = renderer
    .create(
      <Button btnClass="btn-outlined" disabled>
        Outlined Disabled
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is title attribute", () => {
  const tree = renderer
    .create(
      <Button btnClass="btn-outlined" title="Button Title" disabled>
        Title
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is styled button with custom class", () => {
  const tree = renderer
    .create(
      <Button
        style={{ fontStyle: "italic", padding: "20px" }}
        className="text-green-600"
      >
        Styled Button
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
