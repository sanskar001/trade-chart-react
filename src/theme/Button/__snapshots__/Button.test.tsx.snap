// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is button with onclick handle 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Default
</button>
`;

exports[`renders correctly when there is default button 1`] = `
<button
  className="btn"
  disabled={false}
  type="button"
>
  Default
</button>
`;

exports[`renders correctly when there is disabled outlined button 1`] = `
<button
  className="btn btn-outlined"
  disabled={true}
  type="button"
>
  Outlined Disabled
</button>
`;

exports[`renders correctly when there is outlined button 1`] = `
<button
  className="btn btn-outlined"
  disabled={false}
  type="button"
>
  Outlined
</button>
`;

exports[`renders correctly when there is styled button with custom class 1`] = `
<button
  className="btn text-green-600"
  disabled={false}
  style={
    Object {
      "fontStyle": "italic",
      "padding": "20px",
    }
  }
  type="button"
>
  Styled Button
</button>
`;

exports[`renders correctly when there is submit button 1`] = `
<button
  className="btn"
  disabled={false}
  type="submit"
>
  Submit
</button>
`;

exports[`renders correctly when there is title attribute 1`] = `
<button
  className="btn btn-outlined"
  disabled={true}
  title="Button Title"
  type="button"
>
  Title
</button>
`;
